libraries/anvill_passes/src/BrightenPointerOperations.cpp

```diff
@@ -294,40 +300,43 @@ PointerLifter::visitBitCastInst(llvm::BitCastInst &inst) {
     auto [new_var_type, opt_success] =
         visitInferInst(pointer_inst, inst.getDestTy());
     if (opt_success) {
-      ReplaceAllUses(&inst, new_var_type);
+      if (new_var_type->getType() == inst.getType()) {
+        ReplaceAllUses(&inst, new_var_type);
+      }
       return {new_var_type, true};
     }
-//    DLOG(WARNING) << "Bitcast: Failed to propagate info with pointer_inst: "
-//                  << remill::LLVMThingToString(pointer_inst) << "\n";
+    //    DLOG(WARNING) << "Bitcast: Failed to propagate info with pointer_inst: "
+    //                  << remill::LLVMThingToString(pointer_inst) << "\n";
     return {&inst, false};
   }
   // TODO (Carson) make sure in the visitor methods that we dont assume inferred
   // type is the return type, we can fail!
-//  DLOG(WARNING) << "BitCast, unknown operand type: "
-//                << remill::LLVMThingToString(inst.getOperand(0)->getType());
+  //  DLOG(WARNING) << "BitCast, unknown operand type: "
+  //                << remill::LLVMThingToString(inst.getOperand(0)->getType());
   return {&inst, false};
 }
 
 // Use this to flatten a gep
 // Replace all uses of the original gep with the return value (final flattened)
 // Remove spurious instructions with dead code removal.
-llvm::Value *
-PointerLifter::flattenGEP(llvm::GetElementPtrInst *gep) {
+llvm::Value *PointerLifter::flattenGEP(llvm::GetElementPtrInst *gep) {
+
   // FIXME (Carson), delete once we can fix the TODO with getting indexed types
```

Comment:
Can you file an issue with more about what this FIXME relates to? If so, update, the comment to reference the issue number.